src/build_multiversion.zig

```diff
@@ -0,0 +1,839 @@
+    ) |tigerbeetle_current, past_versions, header_name| {
+        const current_checksum = try checksum_file(
+            shell,
+            tigerbeetle_current,
+            multiversion_binary_size_max,
+        );
+
+        var header = multiversioning.MultiversionHeader{
+            .current_release = (try multiversioning.Release.parse(vrs_options.release.?)).value,
+            .current_checksum = current_checksum,
+            .current_flags = .{
+                .debug = true, // FIXME
+                .visit = true,
+            },
+            .past = past_versions.past_releases,
+            .checksum_binary_without_header = checksum_binary_without_header,
+            .current_release_client_min = (try multiversioning.Release.parse(
+                vrs_options.release_client_min.?,
+            )).value,
+            .current_git_commit = try git_sha_to_binary(&vrs_options.git_commit.?),
+        };
+        header.checksum_header = header.calculate_header_checksum();
+
+        try shell.cwd.writeFile(.{
+            .sub_path = header_name,
+            .data = std.mem.asBytes(&header),
+            .flags = .{ .exclusive = true },
+        });
+    }
+
+    try macos_universal_binary_build(shell, options.output, &.{
+        .{
+            .cpu_type = std.macho.CPU_TYPE_ARM64,
+            .cpu_subtype = std.macho.CPU_SUBTYPE_ARM_ALL,
+            .path = options.tigerbeetle_current_aarch64,
+        },
+        .{
+            .cpu_type = std.macho.CPU_TYPE_X86_64,
+            .cpu_subtype = std.macho.CPU_SUBTYPE_X86_64_ALL,
+            .path = options.tigerbeetle_current_x86_64,
+        },
+        .{
+            .cpu_type = @intFromEnum(section_to_macho_cpu.tb_mvb_aarch64),
+            .cpu_subtype = 0x00000000,
+            .path = sections.aarch64.body,
+        },
+        .{
+            .cpu_type = @intFromEnum(section_to_macho_cpu.tb_mvh_aarch64),
+            .cpu_subtype = 0x00000000,
+            .path = sections.aarch64.header,
+        },
+        .{
+            .cpu_type = @intFromEnum(section_to_macho_cpu.tb_mvb_x86_64),
+            .cpu_subtype = 0x00000000,
+            .path = sections.x86_64.body,
+        },
+        .{
+            .cpu_type = @intFromEnum(section_to_macho_cpu.tb_mvh_x86_64),
+            .cpu_subtype = 0x00000000,
+            .path = sections.x86_64.header,
+        },
+    });
+}
+
+fn make_deterministic(shell: *Shell, options: struct {
+    llvm_objcopy: []const u8,
+    source: []const u8,
+    output: []const u8,
+}) !u128 {
+    // Copy the object using llvm-objcopy before taking our hash. This is to ensure we're
+    // round trip deterministic between adding and removing sections:
+    // `llvm-objcopy --add-section ... src dst_added` followed by
+    // `llvm-objcopy --remove-section ... dst_added src_back` means
+    // checksum(src) == checksum(src_back)
+    // Note: actually don't think this is needed, we could assert it?
```

Comment:
To expand on this - this no-op copy was needed when I was initially using GNU objcopy, but I don't think it's needed for LLVM objcopy. We could instead actually add and remove a section here, and check the result is deterministic - which would assert the behaviour we actually want!

Reply:
It seems like the result isn't actually deterministic?

https://github.com/tigerbeetle/tigerbeetle/commit/ec90ba43aea7425239d5c436eb94667d57614c49

Reply:
Hm - can you try with the dummy objcopy? Maybe it still is needed! (The end result should definitely be deterministic, since we verify that modified binary against the `current_checksum` which was taken before any other operations, but after the dummy objcopy). 

It could also be that the dummy objcopy was needed only for debug, or something along those lines.

Reply:
Checked debug and release manually, in both cases the results looks like this: 

```
0468184187d6c475de4245d7bb05c23cd87cc54b20a165c55931e9b40dfc7665  tigerbeetle
fa4f97f9d9c62ed5f686e760232b2c6c7a5806a0f538bbe898668f9f0156527b  tigerbeetle-add-section
046bf770b824c0ec7faf6b8bdec03cb3c015311e2026650875787335ce905905  tigerbeetle-add-section-remove-section
046bf770b824c0ec7faf6b8bdec03cb3c015311e2026650875787335ce905905  tigerbeetle-noop
```

That is, no-op copy is equal to adding&removing the section, but is different from tigerbeetle we get directly out of the build.